app/export.go

```diff
@@ -167,6 +175,46 @@ func (a *App) ExportAllUsers(writer io.Writer) *model.AppError {
 	return nil
 }
 
+func (a *App) ExportAllDirectChannels(writer io.Writer) *model.AppError {
+	afterId := strings.Repeat("0", 26)
+	for {
+		result := <-a.Srv.Store.Channel().GetAllDirectChannelsForExportAfter(1000, afterId)
+
+		if result.Err != nil {
+			return result.Err
+		}
+
+		channels := result.Data.([]*model.DirectChannelForExport)
+
+		if len(channels) == 0 {
+			break
+		}
+
+		for _, channel := range channels {
+			afterId = channel.Id
+
+			// Skip deleted.
+			if channel.DeleteAt != 0 {
+				continue
+			}
+
+			members := strings.Split(channel.Usernames, ",")
+			channel.Members = &members
+
+			if len(members) == 1 {
```

Comment:
I'm assuming this is to omit self direct messages? Can you elaborate on what is needed to support this? Seems like it ought to "just work", but I'm not super familiar with our export process.

Reply:
@lieut-data the Bulk Import doesn't support self-DM channels currently, so I suggested leaving them out of the export until we've implmeneted it in import too.